simulation/policies.py

```diff
@@ -537,37 +537,26 @@ def su_update_sentiment_when_proposal_becomes_failed_or_completed(params, step,
         policy_output_passthru = s["policy_output"]
 
         report = {}
-        for idx in policy_output_passthru["failed"]:
-            for participant_idx, proposal_idx, _ in find_in_edges_of_type_for_proposal(network, idx, "support"):
-                edge = network.edges[participant_idx, proposal_idx]
-                if edge["affinity"] == 1:
-                    sentiment_old = network.nodes[participant_idx]["item"].sentiment
-                    sentiment_new = sentiment_old + config.sentiment_bonus_proposal_becomes_failed
-                    sentiment_new = 0 if sentiment_new < 0 else sentiment_new
-                    network.nodes[participant_idx]["item"].sentiment = sentiment_new
-
-                    report[participant_idx] = {
-                        "proposal_idx": proposal_idx,
-                        "sentiment_old": sentiment_old,
-                        "sentiment_new": sentiment_new,
-                        "status": "failed"
-                    }
-        
-        for idx in policy_output_passthru["succeeded"]:
-            for participant_idx, proposal_idx, _ in find_in_edges_of_type_for_proposal(network, idx, "support"):
-                edge = network.edges[participant_idx, proposal_idx]
-                if edge["affinity"] == 1:
-                    sentiment_old = network.nodes[participant_idx]["item"].sentiment
-                    sentiment_new = sentiment_old + config.sentiment_bonus_proposal_becomes_completed
-                    sentiment_new = 1 if sentiment_new > 1 else sentiment_new
-                    network.nodes[participant_idx]["item"].sentiment = sentiment_new
-
-                    report[participant_idx] = {
-                        "proposal_idx": proposal_idx,
-                        "sentiment_old": sentiment_old,
-                        "sentiment_new": sentiment_new,
-                        "status": "completed"
-                    }
+        proposal_status = ["failed", "succeeded"]
+        sentiment_deltas = [config.sentiment_bonus_proposal_becomes_failed, config.sentiment_bonus_proposal_becomes_completed]
+        for status, delta in zip(proposal_status, sentiment_deltas):
+            for idx in policy_output_passthru[status]:
+                for participant_idx, proposal_idx, _ in find_in_edges_of_type_for_proposal(network, idx, "support"):
+                    edge = network.edges[participant_idx, proposal_idx]
+                    # Update the participant sentiment if he/she is the proposal creator (affinity = 1)
```

Comment:
It's dangerous to rely on affinity=1 to determine proposal creator.
What we need is an additional field, which would obviously add to the complexity of the code - maybe it's time to actually convert the edge into a class. Not sure yet.

Reply:
do you want me to give it a try on it or should I just add an "isCreator" property when participant-to-proposal edges are being setup?

Reply:
The problem is when we add a `isCreator` property in the edge setup code, we will have to (as humans) remember to update every piece of code that deals with support edges. For now it seems ok to do it, but it's always like that in the beginning, then the technical debt grows and grows.

Because networkx stores such things as dict attributes, linter can't possibly know about them (it doesn't know what attributes your dict can possibly have). We could probably use a NamedTuple instead of a class here to represent the edge.

Reply:
Agree! Maybe we can leave this PR as is now and tackle this new feature in a separate PR